content/PHPDataObjects/DataObjects.q.md

```diff
@@ -0,0 +1,1241 @@
+    $password = 'Admin01';
+
+    try {
+        $conn = new PDO("pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password");
+        if ($conn) {
+            echo "Successfully connected to the database. ";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection."; 
+    }
+?>
+```
+
+```php
+// student.php
+<?php
+    require_once("connection.php");
+
+    try {
+        $stmt = $conn->query("INSERT INTO students (first_name, last_name, birth_date, gender) VALUES ('John', 'Smith', '1999-02-10', 'Male')");
+        if (!$stmt) {
+            throw new Exception("Unable to insert values into the table.");
+        }
+        echo "Successfully inserted the values into the table.";
+    } catch (Exception $e) {
+        echo $e->getMessage();
+    }
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`
+
+- It prints `Successfully connected to the database.` and `Unable to insert values into the table.`.
+
+- It prints `Successfully connected to the database.` and `Successfully inserted the values into the table.`.
+
+
+/// type=MS, answer=[4,5]
+
+Which statements correctly describe the code on line 9 of `connection.php`?
+
+- It evaluates the `$conn` object.
+
+- It sets the `PDO` class to the `PDO_PGQSL DSN` statement
+
+- It assigns the `PDO_PGQSL DSN` statement to the `PDO` class.
+
+- It creates the `$conn` object as an instance of the `PDO` class.
+
+- It passes the `PDO_PGQSL DSN` statement as an argument of the `PDO` class.
+
+
+/// type=MS, answer=[1,2,3,5]
+
+What values are inserted by the statement `INSERT INTO students (first_name, last_name, birth_date, gender) VALUES ('John', 'Smith', '1999-02-10', 'Male')")` on line 5?
+
+- `Male`
+
+- `John`
+
+- `Smith`
+
+- `first_name`
+
+- `1999-02-10`
+
+
+/// type=SS, answer=[5]
+
+In `student.php`, what does the query `INSERT INTO students (first_name, last_name, birth_date, gender) VALUES ('John', 'Smith', '1999-02-10', 'Male')")` do on line 5?
+
+- It modifies multiple tables.
+
+- It adds three columns in the `students` table.
+
+- It inserts one column into the `students` table.
+
+- It modifies the values `John`, `Smith`, `1999-02-10`, and `Male` in the `students` table.
+
+- It adds a new row that contains the values `John`, `Smith`, `1999-02-10`, and `Male` in the `students` table.
+
+
+/// type=MS, answer=[2,3,4,5]
+
+Which statements correctly describe the code on line 5 of `student.php`?
+
+- It connects to the PostgreSQL database.
+
+- It calls the `query()` method of the `$conn` object.
+
+- It returns the result set as a `PDOStatement` object.
+
+- It assigns the `PDOStatement` object to the variable `$stmt`.
+
+- It executes the argument `INSERT INTO students (first_name, last_name, birth_date, gender) VALUES ('John', 'Smith', '1999-02-10', 'Male')")` in the `query()` method.
```

Comment:
Typo:

Please remove __`")`__ from the `INSERT INTO` statement here.... Thanks.